### PR TITLE
[CCXDEV-3462] Add error_key to vote/feedback related tables

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

The database only stores the rule_id when a user enable, disable, gives feedback or vote a rule hit result. This results on that a rule with several possible error keys can only be stored as a whole, not possible to enable/disable an specific hit on that rule.

The commit adds support on the database to store the error_key and enables the related endpoints to store and retrieve data from the updated tables.

Fixes insights-results-smart-proxy#340 

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## Testing steps

Regular CI

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: server/router_utils.go [hunk 1]
```diff
@@ -197,29 +197,34 @@ func (server *HTTPServer) readClusterRuleUserParams(
...
-// readClusterRuleParams gets cluster_name and rule_id from current request
+// readClusterRuleParams gets cluster_name, rule_id and error_key from current request
 func (server *HTTPServer) readClusterRuleParams(
 	writer http.ResponseWriter, request *http.Request,
-) (types.ClusterName, types.RuleID, bool) {
+) (types.ClusterName, types.RuleID, types.ErrorKey, bool) {
 	clusterID, successful := readClusterName(writer, request)
 	if !successful {
-		return "", "", false
+		return "", "", "", false
```
Comment (tisnik) [inline server/router_utils.go:206]: the right time to use named return values, WDYT?

File: server/router_utils.go [hunk 2]
```diff
@@ -197,29 +197,34 @@ func (server *HTTPServer) readClusterRuleUserParams(
...
 	}
 
 	ruleID, successful := readRuleID(writer, request)
 	if !successful {
-		return "", "", false
+		return "", "", "", false
```
Comment (tisnik) [inline server/router_utils.go:211]: dtto

File: server/router_utils.go [hunk 3]
```diff
@@ -197,29 +197,34 @@ func (server *HTTPServer) readClusterRuleUserParams(
...
 	clusterExists, err := server.Storage.DoesClusterExist(clusterID)
 	if err != nil {
 		handleServerError(writer, err)
-		return "", "", false
+		return "", "", "", false
 	}
 	if !clusterExists {
 		handleServerError(writer, &types.ItemNotFoundError{ItemID: clusterID})
-		return "", "", false
+		return "", "", "", false
```
Comment (tisnik) [inline server/router_utils.go:226]: dtto

**Review by tisnik — APPROVED** [review]
Very nice one! I just found the function with four return values quite unreadable ATM, that's all.

File: server/router_utils.go (hunk 1 above)
Comment (joselsegura) [reply server/router_utils.go]: you are right, new commit incoming

**Review by tisnik — APPROVED** [review]
LGTM!!!

**Review by Bee-lee — APPROVED** [review]
Looks good to me! Very nice.

We just have to make sure it isn't displayed weird in OCM UI, a user can run into a situation where he disables a rule and then the cluster hits a different error key of the same rule and it will pop back again since we're now selecting by `error_key` too.
Since I believe we're showing a general message in the lists, and also we're planning to accommodate for "general" summary/resolution/reason as well as ones specific to the error keys, it could look like a bug by having the rule duplicated or seemingly re-enabled before going to the details of them.

**Comment by joselsegura** [comment]
> Looks good to me! Very nice.
> 
> We just have to make sure it isn't displayed weird in OCM UI, a user can run into a situation where he disables a rule and then the cluster hits a different error key of the same rule and it will pop back again since we're now selecting by `error_key` too.
> Since I believe we're showing a general message in the lists, and also we're planning to accommodate for "general" summary/resolution/reason as well as ones specific to the error keys, it could look like a bug by having the rule duplicated or seemingly re-enabled before going to the details of them.

Yes, we need to check it very carefully. I suppose that different error keys will be shown differently in OCM UI (and TBH most of the rules that we are running in the pipeline only have 1 error key)

**Comment by joselsegura** [comment]
I'm closing this PR at the moment because I just realised that I need to update one more table, which its database schema was not touched from the migration part

**Comment by joselsegura** [comment]
Reopened the PR. Updates on missing tables and proper manually testing about migration in Postgres DB

**Review by Bee-lee — APPROVED** [review]
LGTM still

**Review by dpensi — APPROVED** [review]

**Review by epapbak — APPROVED** [review]
lgtm

**Pull request merged**